### PR TITLE
Order Creation: Connect status picker

### LINKED ISSUE #5406: Order Creation: Order Status section UI
When a new order is created, the default order status is `pending`. We need a section at the top of the New Order screen showing the current date and editable order status (like in Order Details). When the `Edit` button is tapped, a list opens to select a new status.

### Tasks

Feel free to divide these into separate subtasks:

- [ ] Display current date
- [ ] Display order status (`pending` by default)
- [ ] Select different order status from a list

Order Status section|Order Status list
-|-
iOS-CreateOrder-Status|iOS-CreateOrder-Edit Discount

### PR DESCRIPTION
**Pull request by ealeksandrov**
Closes: https://github.com/woocommerce/woocommerce-ios/issues/5406.

## Description

This PR connects navigation from NewOrder view to StatusList so user can pick different status.

## Changes

- Adds SwiftUI wrapper for `OrderStatusListViewController`.
- Replaces separate `var orderStatus` with observation of `orderDetails` to keep `StatusBadgeViewModel` in sync with model.
- Adds navigation to `OrderStatusList` and its callback handling to update `orderDetails.status`.
- Adds project folders for StatusSection and FlowCoordinator under "Order Creation".

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (Tap "Create order" in the bottom sheet if that appears.)
4. Verify that status section appears with "Pending payment" status.
5. Tap "Edit" button to display status picker.
6. Select different status and tap "Apply".
7. Verify that status section updates to display selected status.

### Considerations

This change will enable "Create" button, but its navigation and saving any of order data is not working now.

## Screenshots

statuses list|updated status
--|--
Simulator Screen Shot|Simulator Screen Shot

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift [hunk 1]
```diff
@@ -182,11 +181,16 @@ private extension NewOrderViewModel {
...
 
-    func setInitialOrderStatus() {
-        guard let pendingSiteStatus = currentSiteStatuses.first(where: { $0.status == .pending }) else {
-            return
-        }
-
-        orderStatus = pendingSiteStatus
+    /// Updates status badge viewmodel based on status order property.
+    ///
+    func configureStatusBadgeViewModel() {
```
Comment (rachelmcr) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift:186]: Should we have a unit test to check that the status badge view model is updated as expected?

**Review by rachelmcr — APPROVED** [review]
Looks great! I just have one question below about unit testing, but this works well and the connection to `OrderDetails` is nicely done. That should make it fairly straightforward to do the remote request. 👍

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift (hunk 1 above)
Comment (ealeksandrov) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift:186]: Good catch! Added in f1db99e.

**Comment by ealeksandrov** [comment]
Thanks for review Rachel! Added unit test and will merge after it's green.

**Pull request merged**